enhancements/generic-constraints.md

```diff
@@ -0,0 +1,327 @@
+
+For example, the bundle properties map (which is an input to the resolver) is:
+
+```yaml
+properties:
+  - property:
+      type: sushi
+      value: salmon
+  - property:
+      type: soup
+      value: miso
+  - property:
+      type: olm.gvk
+      value:
+        group: olm.coreos.io
+        version: v1alpha1
+        kind: bento
+```
+
+The example constraint is specified to have a dependency for a custom property `sushi` with value `salmon`:
+
+```yaml=
+type: olm.constraint
+value:
+  message: "require to have the property `sushi` with value `salmon`"
+  CEL:
+    rule: "properties.exists(p, p.type == 'sushi' && p.value == 'salmon')"
+```
+
+The cel expression will search for a property with the type `sushi` and the value `salmon` within the bundle property map and return `true` or `false`.
+
+#### Compound Constraint
+
+The CEL [syntax](https://github.com/google/cel-spec/blob/master/doc/langdef.md#syntax) supports a wide range of operators including logic operator such as AND and OR. As a result, a single CEL expression can have multiple rules for multiple conditions that are linked together by logic operators. These rules are evaluated against a dataset of multiple different properties from a bundle or any given source and the output is solved into a single bundle or operator that satisfies all of those rules within a single constraint. For example:
+
+```json=
+{
+  "type": "olm.constraint",
+  "value": {
+      "message": 'require to have "certified" and "stable" properties',
+      "CEL": {
+          "rule": 'properties.exists(p, p.type == "certified") && properties.exists(p, p.type == "stable")'
+      }
+  }
+}
+```
+
+The expression in `rule` has two requirements that are linked in an AND (&&) logical operator. In order for that expression to be true, both requirements must be satisfied. Therefore, the resolved operator will be a single bundle that has both certified and stable properties. This concept represents the compound constraint.
+
+### Graduation Criteria
+
+This generic constraint is currently scheduled to be available in the next release with a set of supported functionalities. More functionalities will be added to the feature if needed in the subsequent releases.
+
+Alternatively, it is possible to wrap this feature under feature flag which means it is disabled by default. As a result, the feature can be a dev-preview/tech-preview feature and begins to gather users' feedbacks.
+
+As the feature is being used and evaluated in production, changes can be made to address feedbacks and improve overall usefulness and user experience. Then, the feature flag can be removed and the feature is enabled by default moving forward. At the time, the feature is GA.
```

Comment:
Did we make a final decision on tech preview vs GA?

Reply:
As far as I know, it is gonna be GA right away especially if we need it for the cluster constraint.